sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -54,14 +54,6 @@ trait BaseLimitExec extends UnaryExecNode with CodegenSupport {
   val limit: Int
   override def output: Seq[Attribute] = child.output
 
-  // Do not enable whole stage codegen for a single limit.
-  override def supportCodegen: Boolean = child match {
-    case plan: CodegenSupport => plan.supportCodegen
-    case _ => false
```

Comment:
This is wrong, we may have more operators above `Limit`, so it's not a single `Limit`.